db/large_data_handler.cc

```diff
@@ -46,12 +46,24 @@ large_data_handler::large_data_handler(uint64_t partition_threshold_bytes, uint6
         partition_threshold_bytes, row_threshold_bytes, cell_threshold_bytes, rows_count_threshold);
 }
 
-future<bool> large_data_handler::maybe_record_large_partitions(const sstables::sstable& sst, const sstables::key& key, uint64_t partition_size) {
+future<bool> large_data_handler::maybe_record_too_many_rows(const sstables::sstable& sst, const sstables::key& key, uint64_t partition_size, uint64_t rows_count) {
+    assert(running());
+    if (__builtin_expect(rows_count > _rows_count_threshold, false)) {
+        return with_sem([&sst, &key, partition_size, rows_count, this] {
+            return record_large_partitions(sst, key, partition_size, rows_count);
+        }).then([] {
+            return true;
+        });
+    }
+    return make_ready_future<bool>(false);
+}
+
+future<bool> large_data_handler::maybe_record_large_partitions(const sstables::sstable& sst, const sstables::key& key, uint64_t partition_size, uint64_t rows_count) {
```

Comment:
If we have `row_count` at hand when calling this function, we can fold `maybe_record_too_many_rows` above into `maybe_record_large_partitions` that will check both threshold.
I don't think it's justified to have two "maybe" functions that are called in succession anyhow with exactly the same args.